proc/proc.go

```diff
@@ -0,0 +1,249 @@
+// proc offers control over arbitrary processes.
+// proc depends heavily on os/exec.
+// If/when proc needs to do resource control & isolation
+// its probable that os/exec will need to be replaced with
+// either the os package itself or syscall.
+package proc
+
+// Fingers crossed that its easier than rewriting os.ForkExec
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"sync"
+	"time"
+)
+
+const outputReadRefreshInterval = time.Millisecond * 10
+
+// Proc is a BitBox process.
+// Stderr and Stdout are dumped to temp files on disk.
+type Proc struct {
+	// TODO: there is 99% chance we need a synchronization primitive at this level
```

Comment:
can you explain why we might/not need the sync primitive?

Reply:
Okay. I've protected `cmd.Wait` with a mutex. Anything else you feel is pending here @NajiObeid?